website/project/model.py

```diff
@@ -2684,23 +2720,18 @@ def _initiate_embargo(self, user, end_date, for_existing_registration=False, sav
         :param user: User who initiated the retraction
         :param end_date: Date when the registration should be made public
         """
-
-        embargo = Embargo()
-        embargo.initiated_by = user
-        embargo.for_existing_registration = for_existing_registration
-        # Convert Date to Datetime
-        embargo.end_date = datetime.datetime.combine(end_date, datetime.datetime.min.time())
-
+        embargo = Embargo(
+            initiated_by=user,
+            end_date=datetime.datetime.combine(end_date, datetime.datetime.min.time()),
+            for_existing_registration=for_existing_registration
+        )
+        embargo.save()
+        self.embargo = embargo
+        self.save()
```

Comment:
@sloria For embargoes, retractions, and registration approvals the Node's foreign field to that record must be set (L2730). ModularODM needs a pk to set the foreign field (L2728). 

In particular add_authorizer has some logic that creates stateful versions of approval/rejection urls (https://github.com/CenterForOpenScience/osf.io/pull/3919/files#diff-b4b50a4acab364147a53259fa3fe9b3fR3090). Before Archiver was responsible for stashing these urls (https://github.com/CenterForOpenScience/osf.io/pull/3919/files#diff-bc9bf8c80ec1898eafb640e6b97d1460L462; which to me was far too tight a coupling of unrelated code). 

The second save here makes sure the embargo get's saved after being mutated by add_authorizer (rather than saving after each mutation, i.e. ```add_authorizer(user, save=True)```).

Agreed this is a bit awkward, but I'm leaning toward just removing the save kwarg from the signature (and adding docs) unless you have a cleaner approach.